src/common/util/arrow.h

```diff
@@ -0,0 +1,31 @@
+/** Copyright 2020-2022 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef SRC_COMMON_UTIL_ARROW_H_
+#define SRC_COMMON_UTIL_ARROW_H_
+#include <utility>
+
+namespace vineyard {
+#define RETURN_ON_ARROW_ERROR_AND_ASSIGN(lhs, expr)           \
+  do {                                                        \
+    auto result = (expr);                                     \
+    if (!result.status().ok()) {                              \
+      return ::vineyard::Status::ArrowError(result.status()); \
+    }                                                         \
+    lhs = std::move(result).ValueOrDie();                     \
+  } while (0)
+}  // namespace vineyard
+
```

Comment:
Could you please remove the same macro definition in `modules/basic/ds/arrow_utils.h` and include this header in that file?

The two macro looks identical and there's no reason for such a duplication.